developing_packages/editable_packages.rst

```diff
@@ -0,0 +1,250 @@
+from the conan cache.
+
+Conan package recipes define a package "layout" in their ``package_info()`` methods. The default one,
+if nothing is specified is equivalent to:
+
+.. code-block:: python
+
+    def package_info(self):
+        # default behavior, doesn't need to be explicitly defined in recipes
+        self.cpp_info.includedirs = ["include"]
+        self.cpp_info.libdirs = ["lib"]
+        self.cpp_info.bindirs = ["bin"]
+        self.cpp_info.resdirs = ["res"]
+
+That means that conan will use the path ``path/to/local/dev/libcool/include`` for locating the headers of
+the ``cool`` package, the ``path/to/local/dev/libcool/lib`` to locate the libraries of the package, and so on.
+
+That might not be very useful, as typically while editing the source code and doing incremental builds, the
+development layout is different from that final "package" layout. While it is possible to run a
+:command:`conan package` local command to execute the packaging in the user folder, and that will achieve that
+final layout, that is not very elegant. Conan provides several ways to customize the layout for editable packages.
+
+Editable packages layouts
+-------------------------
+
+The custom layout of a package while it is in editable mode can be defined in different ways:
+
+Recipe defined layout
+++++++++++++++++++++++
+
+A recipe can defined a custom layout when it is not living in the local cache, in its ``package_info()`` method,
+something like:
+
+.. code-block:: python
+
+    from conans import ConanFile
+    
+    class Pkg(ConanFile):
+        settings = "build_type"
+        def package_info(self):
+            if not self.in_local_cache:
+                d = "include_%s" % self.settings.build_type
+                self.cpp_info.includedirs = [d.lower()]
+
+That will map the include directories to ``path/to/local/dev/libcool/include_debug`` when working with ``build_type=Debug``
+conan setting, and to ``path/to/local/dev/libcool/include_release`` if ``build_type=Release``. In the same way, other 
+directories (libdirs, bindirs, etc) can be customized, with any logic, different for different OS, build systems, etc.
+
+.. code-block:: python
+
+    from conans import ConanFile
+    
+    class Pkg(ConanFile):
+        settings = "os", "compiler", "arch", "build_type"
+        def package_info(self):
+            if not self.in_local_cache:
+                if self.settings.compiler == "Visual Studio":
+                    # NOTE: Use the real layout used in your VS projects, this is just an example
+                    self.cpp_info.libdirs = ["%s_%s" % (self.settings.build_type, self.settings.arch)]
+
+That will define the libraries directories to ``path/to/local/dev/libcool/Release_x86_64``, for example.
+That is only an example, the real layout used by VS would be different.
+
+Layout files
++++++++++++++
+
+Layout files have the following syntax:
```

Comment:
Would be nice to have some transition phrase or introduction of layout files.
For example:
"Instead of changing recipes it's possible to define layout in a separate file. This is especially useful if you have a large number of components with similar structure".

Reply:
Check here: https://github.com/conan-io/docs/pull/1046